apps/src/applab/appStorage.js

```diff
@@ -233,15 +232,14 @@ var handleDeleteRecord = function(tableName, record, onSuccess, onError) {
     return;
   }
   if (this.status === 404) {
-    onError('error deleting record: could not find record id ' + record.id +
-        ' in table ' + tableName);
+    onSuccess(false);
     return;
   }
   if (this.status < 200 || this.status >= 300) {
     onError('error deleting record: unexpected http status ' + this.status);
     return;
   }
-  onSuccess();
+  onSuccess(true);
```

Comment:
Also, it's worth noting that we expose an undocumented `onError` parameter to the applab user so that they can handle exceptional cases if they really want to. So now we'll have onComplete and onError. It's possible that onError will go away some time in the future, but that's not clear from the current email discussion on this feature.

I still think it's ok to rename onSuccess to onComplete since it gives a better idea of whats going on regardless of this issue. So just FYI in case that changes your opinion of anything.